CODEOWNERS: Add sig-foundations

joamaki opened the pull request:
Assign pkg/{hive,statedb,promise,stream} and daemon/cmd/cells.go to sig-foundations.

joestringer commented on CODEOWNERS:
```diff
@@ -175,6 +175,8 @@
 #   control plane layers that populate most eBPF maps; account for endianness
 #   and system architecture impacts on the datapath code.
+# - @cilium/sig-foundations:
+#   Maintain the core libraries and frameworks.
```
In general the idea behind these descriptions is to help provide some guidelines about special "gotchas" or context-specific knowledge that a reviewer can use in order to provide useful feedback on PRs that modify these code areas.

I wonder if we can expand on this a bit, for instance encouraging careful consideration for common APIs with ideally a single solution that works for all use cases rather than proliferating many APIs for similar use cases.

I also note that the @cilium/sig-agent already covers `Internal architecture, core data structures and daemon startup.`, so an additional question would be, do we need another separate SIG for this or could we fold these files into @cilium/sig-agent ? Is there something materially different between the review of the core agent logic vs. the files that this PR is proposing to be reviewed by a new group?

joamaki replied:
Reworded the description, PTAL. The question regarding adding another SIG we resolved during the community meeting on May 3rd.

pchaigno commented on CODEOWNERS:
```diff
@@ -414,24 +417,24 @@ jenkinsfiles @cilium/ci-structure
 /pkg/fswatcher/ @cilium/sig-datapath @cilium/sig-hubble
 /pkg/health/ @cilium/sig-agent
-/pkg/hive/ @cilium/sig-agent
+/pkg/hive/ @cilium/sig-foundations
```
GitHub reports an issue with the team. Did we ensure it has write access to the repository? That's necessary for reviews for some reason.

joestringer replied:
I've added write access to cilium/cilium for @cilium/sig-foundations .

pchaigno commented on CODEOWNERS:
```diff
@@ -414,24 +417,24 @@ jenkinsfiles @cilium/ci-structure
 /pkg/ipam/allocator/aws/ @cilium/sig-ipam @cilium/aws
 /pkg/ipam/allocator/azure/ @cilium/sig-ipam @cilium/azure
 /pkg/ipam/allocator/clusterpool/ @cilium/sig-ipam @cilium/operator
+/pkg/ipam/ @cilium/sig-ipam
```
This needs to stay as it was or it will overwrite all 4 four subdirectories.

pchaigno commented on CODEOWNERS:
```diff
@@ -414,24 +417,24 @@ jenkinsfiles @cilium/ci-structure
 /pkg/ipmasq @cilium/sig-agent
+/pkg/k8s/apis/cilium.io/client/crds/v2/ @cilium/sig-k8s
+/pkg/k8s/apis/cilium.io/v2/ @cilium/api @cilium/sig-k8s
 /pkg/k8s/ @cilium/sig-k8s
```
Ditto here.

pchaigno commented on CODEOWNERS:
```diff
@@ -445,8 +448,8 @@ jenkinsfiles @cilium/ci-structure
 /pkg/modules/ @cilium/sig-datapath
-/pkg/monitor @cilium/sig-datapath
 /pkg/monitor/api @cilium/api @cilium/sig-datapath
+/pkg/monitor @cilium/sig-datapath
```
Ditto.

pchaigno commented on CODEOWNERS:
```diff
@@ -457,23 +460,26 @@ jenkinsfiles @cilium/ci-structure
-/pkg/proxy/ @cilium/proxy
+/pkg/promise/ @cilium/sig-foundations
 /pkg/proxy/accesslog @cilium/api
+/pkg/proxy/ @cilium/proxy
```
Ditto.

pchaigno commented on CODEOWNERS:
```diff
@@ -457,23 +460,26 @@ jenkinsfiles @cilium/ci-structure
 /pkg/pidfile @cilium/sig-agent
-/pkg/policy @cilium/sig-policy
 /pkg/policy/api/ @cilium/api
+/pkg/policy @cilium/sig-policy
```
Ditto.

pchaigno reviewed (changes requested):
Some of the entries you reordered were in non-alphabetical order because GitHub applies the ownership in the order they are defined. See comment below.

EDIT: Maybe also best to keep those changes in a separate commit.

zacharysarah reviewed (commented):
+1 to consideration for @joestringer's [comment](https://github.com/cilium/cilium/pull/24976/files#r1171691463), otherwise LGTM from a docs perspective.

joamaki committed:
CODEOWNERS: Add sig-foundations

Assign pkg/{hive,statedb,promise,stream} and daemon/cmd/cells.go to
sig-foundations.

Signed-off-by: Jussi Maki <jussi@isovalent.com>

pchaigno reviewed (approved):
One comment above is still relevant (GitHub warning). Otherwise, LGTM.
